src/parse/rules/misc_rules.build_defs

```diff
@@ -537,15 +540,27 @@ def tarball(name, srcs, out=None, deps=None, subdir=None,
                              'explicit flag for "compression"')
     else:
         compression, extension = _COMPRESSION.get(compression, ('-a', ''))
+
+    out = out or name + '.tar' + extension
+    # Unlike the rest of the compressions, gzip preserves the file's (the tar)
+    # timestamp by default, making the output unreproducible.
+    # Hence the compression is done with the -n flag in a separate step
+    if extension == '.gz' or out.endswith('.gz'):
+        archive_cmd = 'tar -c * | gzip -n > $OUT'
+    else:
+        archive_cmd = 'tar %s -cf $OUT *' % compression
+
     build_rule(
         name=name,
         cmd=' && '.join([
             'mkdir -p _tmp/' + subdir,
-            'cp -r $(locations :_%s#files) _tmp/%s' % (name, subdir),
+            'cp -pr $(locations :_%s#files) _tmp/%s' % (name, subdir),
```

Comment:
I'm not 100% sure that they are different any more; at some point we got a bit stricter about chmod'ing things in plz-out/gen. I just recall there being a codepath where I had to write a file and I didn't know what the mode should be, because the cache has lost it by then.

Hopefully it is not a problem, just mentioning it as a potential point of indeterminacy :)